src/App.tsx

```diff
@@ -15,71 +16,81 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+type State = {
+  selectedItem: string | null;
+};
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+export class App extends Component {
+  state: Readonly<State> = {
+    selectedItem: 'Jam',
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  switcher(item: string) {
+    if (item === this.state.selectedItem) {
+      this.setState({ selectedItem: null });
+    } else {
+      this.setState({ selectedItem: item });
+    }
+  }
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  render() {
+    const { selectedItem } = this.state;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+    return (
+      <main className="section container">
+        <h1
+          className={classNames(
+            'title',
+            { 'is-flex is-align-items-center': selectedItem },
+          )}
+        >
+          {selectedItem ? `${selectedItem} is selected` : 'No goods selected'}
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+          {selectedItem && (
+            <>
+              {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+              <button
+                data-cy="ClearButton"
+                type="button"
+                className="delete ml-3"
+                onClick={() => this.switcher(selectedItem)}
+              />
+            </>
+          )}
+        </h1>
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+        <table className="table">
+          <tbody>
+            {goods.map(item => (
+              <tr
+                data-cy="Good"
+                key={item}
+                className={classNames(
+                  { 'has-background-success-light': selectedItem === item },
+                )}
+              >
+                <td>
+                  <button
+                    data-cy={selectedItem === item
+                      ? 'RemoveButton'
+                      : 'AddButton'}
+                    type="button"
+                    className={classNames(
+                      'button',
+                      { 'is-info': selectedItem === item },
```

Comment:
not too sure about boolean class names

Reply:
I don't get it, is-info is a bulma style